### DISCUSSION TITLE
add stream name in stream api response data

### DISCUSSION THREAD

**Discussion by iotv0815** [iptv-org/community#20] [Ideas]
Hello team,
there are many streams have the `tvg-id=""`, this lead to the `channel` value empty in the get stream API call response. 
for example, below is a stream with empty `tvg-id` in a m3u file

#EXTINF:-1 tvg-id="",Foochow TV-1 News
http://live.zohi.tv/video/s10001-FZTV-1/index.m3u8

when making stream API call, https://iptv-org.github.io/api/streams.json
the stream API response data is:
{
    channel: ""
    http_referrer: invalid
    timeshift: invalid
    url: "http://live.zohi.tv/video/s10001-fztv-3/index.m3u8"
    user_agent: invalid
}

only the `url` extracted from the original m3u file without any stream name or id. such stream data is not able to to be associated to any channel, can't be searched either

in its m3u file, although the `tvg-id` is empty, there is a name for the stream. in above example, the stream name is `Foochow TV-1 News`. if this name can be added into the stream API response data then this stream can be linked as `Foochow TV-1 News`, it is also searchable.

so far, there are 1687 streams have empty `channel` value in stream API response.

could someone please update the parser script to add the stream name into stream API call response?
the expected response data:

{
    channel: ""
   name: "Foochow TV-1 News"
    http_referrer: invalid
    timeshift: invalid
    url: "http://live.zohi.tv/video/s10001-fztv-3/index.m3u8"
    user_agent: invalid
}

**Comment by freearhey** [comment]
Added in https://github.com/iptv-org/iptv/pull/25158
